### PR TITLE
Implement subscriptions

### PR DESCRIPTION
**Pull request by gjvoosten**
Add subscriptions and subscription update feeds.

Closes NCI-Agency/anet#1115
Closes NCI-Agency/anet#1344
Closes NCI-Agency/anet#1368
Closes NCI-Agency/anet#1369
Closes NCI-Agency/anet#1370

#### User changes
- User can subscribe to any subscribable object (location, organization, person, position, report, task)
- Subscription updates are shown chronologically on the Home page
- Subscriptions are shown on the user's personal details page

#### Super User changes
- None

#### Admin changes
- None

#### System admin changes
- [ ] anet.yml or anet-dictionary.yml needs change
- [x] db needs migration
- [ ] documentation has changed
- [x] graphql schema has changed

### Checklist
  - [x] Described the user behavior in PR body
  - [x] Referenced/updated all related issues
  - [x] commits follow a `repo#issue: Title` title format and [these 7 rules](https://chris.beams.io/posts/git-commit/)
  - [x] commits have a [clean history](https://epage.github.io/dev/commits/), otherwise PR may be squash-merged
  - [x] Added and/or updated unit tests
  - [ ] Added and/or updated e2e tests
  - [x] Added and/or updated data migrations
  - [ ] Updated documentation
  - [x] Resolved all build errors and warnings
  - [x] Opened debt issues for anything not resolved here

### CODE REVIEW

**Review by anilsonmez-simsoft — CHANGES_REQUESTED** [review]

File: client/src/models/Position.js [hunk 1]
```diff
@@ -95,6 +95,8 @@ export default class Position extends Model {
     name
     type
     status
+    isSubscribed
```
[the remark this answers is not in the document]
Comment (gjvoosten) [reply client/src/models/Position.js:98]: Ah yes, that one wasn't there 2.5 years ago. :wink:

File: client/src/components/MySubscriptionUpdates.js [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+                      [object was deleted]
+                    </LinkTo>
+                  )
+                }
+                if (subscriptionUpdate.isNote) {
+                  linkToUpdatedObject = (
+                    <span>Note on {linkToUpdatedObject}</span>
+                  )
+                }
+                const key = `${subscriptionUpdate.createdAt}:${subscriptionUpdate.updatedObjectType}:${subscriptionUpdate.updatedObjectUuid}`
```
[the remark this answers is not in the document]
Comment (gjvoosten) [reply client/src/components/MySubscriptionUpdates.js:211]: Possibly just a combination of `${subscriptionUpdate.createdAt}:${subscription.uuid}` would suffice.

File: client/src/pages/people/Show.js [hunk 3]
```diff
@@ -270,6 +288,10 @@ const PersonShow = ({ pageDispatchers }) => {
                 />
               )}
 
+              {person.uuid === currentUser.uuid && (
+                <MySubscriptions uuid={person.position.uuid} />
```
[the remark this answers is not in the document]
Comment (gjvoosten) [reply client/src/pages/people/Show.js:292]: No; it was intended to be used for showing subscriptions of any position (not just your own), but that's unimplemented.

File: client/src/models/Position.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (gjvoosten) [reply client/src/models/Position.js:98]: Looks like `Location.allFieldsQuery` should be used in `client/src/pages/locations/Show.js` as well (it isn't right now).

**Review by anilsonmez-simsoft — APPROVED** [review]

File: src/main/java/mil/dds/anet/beans/search/SubscribableObjectSearchQuery.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    return Boolean.TRUE.equals(subscribed);
+  }
+
+  public void setSubscribed(Boolean subscribed) {
+    this.subscribed = subscribed;
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(super.hashCode(), subscribed);
```
Comment (VassilIordanov) [inline src/main/java/mil/dds/anet/beans/search/SubscribableObjectSearchQuery.java:28]: isn't `Objects.hash(super, subscribed);` sufficient?

**Review by VassilIordanov — APPROVED** [review]

File: src/main/java/mil/dds/anet/beans/search/SubscribableObjectSearchQuery.java (hunk 4 above)
Comment (gjvoosten) [reply src/main/java/mil/dds/anet/beans/search/SubscribableObjectSearchQuery.java:28]: Actually, yes, but other places already called it like this, so either we should change them all or leave it like this.

**Review by anilsonmez-simsoft — APPROVED** [review]

**Pull request merged**